### PR TITLE
feat: Add dml_standard_insert sample code.

### LINKED ISSUE #182: Sample: Developers can insert using DML
Sample code:  https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/master/spanner/api/Spanner/Program.cs#L1208

This should involve creating a read-write transaction, and inserting some data into a table using DML, and committing the RW transaction using the `RunTransaction()` function.

### PR DESCRIPTION
**Pull request by coryan**
Fixed #182.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/378)

### CODE REVIEW

File: google/cloud/spanner/samples/samples.cc [hunk 1]
```diff
@@ -134,6 +139,42 @@ void DropDatabase(std::vector<std::string> const& argv) {
...
+        "dml-standard-insert <project-id> <instance-id> <database-id>");
+  }
+
+  // TODO(#377) - cache the client across sample functions.
+  namespace gcs = google::cloud::spanner;
+  gcs::Client client(
+      gcs::MakeConnection(gcs::MakeDatabaseName(argv[0], argv[1], argv[2])));
+
+  //! [START spanner_dml_standard_insert]
+  using google::cloud::future;
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: Unused?

File: google/cloud/spanner/samples/samples.cc [hunk 2]
```diff
@@ -134,6 +139,42 @@ void DropDatabase(std::vector<std::string> const& argv) {
...
+  using google::cloud::StatusOr;
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: Unused?

File: google/cloud/spanner/samples/samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/samples/samples.cc]: Removed.

File: google/cloud/spanner/samples/samples.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/samples/samples.cc]: Removed.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**